programs/ziti-edge-tunnel/windows-service.c

```diff
@@ -472,3 +472,42 @@ DWORD LphandlerFunctionEx(
 
     return 0;
 }
+
+
+BOOL SetPrivilege(HANDLE hToken, LPCTSTR lpszPrivilege, BOOL bEnablePrivilege)
+{
+    LUID luid;
+    BOOL bRet=FALSE;
+
+    if (LookupPrivilegeValue(NULL, lpszPrivilege, &luid))
+    {
+        TOKEN_PRIVILEGES tp;
+
+        tp.PrivilegeCount=1;
+        tp.Privileges[0].Luid=luid;
+        tp.Privileges[0].Attributes=(bEnablePrivilege) ? SE_PRIVILEGE_ENABLED: 0;
+        //
+        //  Enable the privilege or disable all privileges.
+        //
+        if (AdjustTokenPrivileges(hToken, FALSE, &tp, NULL, (PTOKEN_PRIVILEGES)NULL, (PDWORD)NULL))
+        {
+            //
+            //  Check to see if you have proper access.
+            //  You may get "ERROR_NOT_ALL_ASSIGNED".
+            //
+            bRet=(GetLastError() == ERROR_SUCCESS);
+        }
+    }
+    return bRet;
+}
+
+void scm_grant_se_debug() {
+    HANDLE hProcess=GetCurrentProcess();
+    HANDLE hToken;
+
+    if (OpenProcessToken(hProcess, TOKEN_ADJUST_PRIVILEGES, &hToken))
+    {
+        SetPrivilege(hToken, SE_DEBUG_NAME, TRUE);
```

Comment:
can this fail?

Reply:
it returns a BOOL - so presumably it can. If it fails, all that should (*should*) happen is that posture checking will fail on more privileged processes. so we should see DEBUG messages in systems that have posture checking enabled and have a process check configured

Reply:
Might it be nice to also see in the log a WARN (or ERROR) indicating that SetPrivilege failed, and how?